app/controllers/api/v1/posts_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class PostsController < Api::V1::BaseController
+      def index
+        @user = User.find_by(id: params[:user_id])
+        unless @user
+          render json: { message: 'User does not exist' }, status: :not_found
+          return
+        end
```

Comment:
Let's do this with a rescue of ActiveRecord::RecordNotFound. It could even be implemented at the BaseController of the Api module